jetty-websocket/websocket-jetty-api/src/main/java/org/eclipse/jetty/websocket/api/util/WSURI.java

```diff
@@ -104,22 +104,13 @@ public static URI toWebsocket(final URI inputUri) throws URISyntaxException
         Objects.requireNonNull(inputUri, "Input URI must not be null");
         String httpScheme = inputUri.getScheme();
         if ("ws".equalsIgnoreCase(httpScheme) || "wss".equalsIgnoreCase(httpScheme))
-        {
-            // keep as-is
             return inputUri;
-        }
 
+        String afterScheme = inputUri.toString().substring(httpScheme.length());
```

Comment:
`httpScheme` may be `null`, you need a guard like in `Negotiated.toWebSocket()`.

Is it possible to implement this method once and call it from different places?

Reply:
> Is it possible to implement this method once and call it from different places?

Not really as the API does not have dependencies on anything and where it is used in `websocket-core` cannot depend on `websocket-jetty-api`.